DALK.PL_ANALYZER/Models/Matches/MatchesModelViewFactory.cs

```diff
@@ -13,35 +13,52 @@ public class MatchesModelViewFactory
     {
         private FakeDB db;
         public MatchesModelView matchesMV;
-        public MatchesModelViewFactory(int? matchSeasonsId = null, int? matchLeaguesId = null, int? matchTeamsId = null, int? matchGroupId = null, string matchStagesId = null)
+        public MatchesModelViewFactory(RawFilterValues rawParameters)
         {
+            List<GridFilter> allFilters = new List<GridFilter>();
+            MatchesFiltersValues filterValues = new MatchesFiltersValues(rawParameters);
             db = new FakeDB();
 
-            LeaguesSeason leaguesSeason = db.GetLeaguesSeason();
-            TeamsSeason teamSeason = db.GetTeamsSeason();
             Seasons seasons = db.GetSeasons();
             SeasonFilterData defaultSeason = seasons.GetDefaultSeason();
+            GridFilterFactory factory = new GridFilterFactory(seasons.GetSeasons().ToList<IFilterData>(), "Każdy sezon", "matchSeason");
+            factory.SetDefaultFilter(defaultSeason);
+            factory.SetEmptyFilterNotSelected();
+            factory.SetIdByDefault(rawParameters.matchSeasonsId);
+            allFilters.Add(factory.GetGridFilter());
 
-            List<IFilterData> allSeasons = seasons.GetSeasons().ToList<IFilterData>();
-            EmptyFilterDataItem allSeasonFilter = new EmptyFilterDataItem("Każdy sezon", Configs.DEFAULT_FILTER_ICON, false);
-            allSeasons.Add(allSeasonFilter);
+            LeaguesSeason leaguesSeason = db.GetLeaguesSeason();
+            factory = new GridFilterFactory(leaguesSeason.GetLeagueFilterData(defaultSeason).ToList<IFilterData>(), "Każda liga", "matchLeague");
```

Comment:
Here. Dont create another factory pass another paramts into it:
Here, use it like this:
GetFilter(rawParameters.matchSeasonsId`)

Reply:
Good idea, but notice, that this "Factory" class get diffrent params to constructor each time, so i cant just change params in GetFilter, so is it still good idea to make method GetFilter with 5 parameters and empty constructor?

I thought that its better idea to set object data in constructor and have no-param method.

Reply:
OK, but i need not only rawParameters.SOMETHING in method param if i have only one instance of Factory. This method GetFilter would has 4 params - is it still good idea? I thought that its better to set datas in constructor and have no-param method.

Is my understanding correct that purpose is working on 1 instance of factory in that case? But how change configuration of it each time? By calling multi-params method GetFilter? Im not sure.

Reply:
Create 4 methods if you need.